assets/js/atomic/blocks/product-elements/sku/block.tsx

```diff
@@ -9,6 +9,11 @@ import {
 } from '@woocommerce/shared-context';
 import { withProductDataContext } from '@woocommerce/shared-hocs';
 import type { HTMLAttributes } from 'react';
+import {
+	useColorProps,
+	useSpacingProps,
+	useTypographyProps,
+} from '@woocommerce/base-hooks';
```

Comment:
I'm wondering if there's any other way of achieving that or can the hook be used on the Editor side?

The reason I'm asking is these hooks have a dependency on `@wordpress/block-editor` which has a dependency on `@wordpress/components` ([issue](https://github.com/woocommerce/woocommerce-blocks/issues/8071)) and neither of them should be used on the frontend because they're not designed for this and have a performance impact.

To be fair - it's not a blocker, because those hooks are already widely used on a frontend and given the `@wordpress` dependencies are externalized, there's no additional impact by yet another use of hooks (no matter the number of usages, the dependency is loaded just once). Having said that, I'm just raising this for awareness and maybe there's an easy way of avoiding adding another use of that (don't know that 😅 ).

Reply:
TBH, I was thinking about writing a P2 post about it. Ideally, we shouldn't need useColorProps, useSpacingProps, useTypographyProps hooks. 

Instead [here](https://github.com/woocommerce/woocommerce-blocks/blob/9e10039b4471566084ed3b99b808db92e3b713eb/assets/js/atomic/blocks/product-elements/save.js#L15) in save.js, we should return following instead:
```
export const Save = ( { attributes }: Props ): JSX.Element => {
	return (
		<div
			{ ...useBlockProps.save( {
				className: classnames( 'is-loading', attributes.className ),
			} ) }
		/>
	);
};
```

AFAIK `useBlockProps` hooks will generate `className` & `styles`, which we generate using these hooks(useColorProps, useSpacingProps, useTypographyProps). 
But I believe as we need to support product elements in `All products` block, we are using these hooks. 

@Aljullu Is my understanding correct? 🤔

Reply:
> If I'm not wrong, we have some other product element blocks that depend on those hooks as well.

That's true and as I wrote earlier "...I'm just raising this for awareness...". So I'm resolving the thread 👍